modules/core/include/opencv2/core/hal/intrin_neon.hpp

```diff
@@ -1045,9 +1045,24 @@ OPENCV_HAL_IMPL_NEON_BIN_FUNC(v_float64x2, v_absdiff, vabdq_f64)
 /** Saturating absolute difference **/
 inline v_int8x16 v_absdiffs(const v_int8x16& a, const v_int8x16& b)
 { return v_int8x16(vqabsq_s8(vqsubq_s8(a.val, b.val))); }
-inline v_int16x8 v_absdiffs(const v_int16x8& a, const v_int16x8& b)
+static inline v_int16x8 v_absdiffs(const v_int16x8& a, const v_int16x8& b)
 { return v_int16x8(vqabsq_s16(vqsubq_s16(a.val, b.val))); }
 
+static inline v_float32x4 v_absdiffs(const v_float32x4& a, const v_float32x4& b)
+{
+    return v_float32x4(vabsq_f32(vsubq_f32(a.val, b.val)));
+}
+
+static inline v_uint8x16 v_absdiffs(const v_uint8x16& a, const v_uint8x16& b)
+{
+    return v_max(a, b) - v_min(a, b);
+}
+
+static inline v_uint16x8 v_absdiffs(const v_uint16x8& a, const v_uint16x8& b)
+{
+    return v_max(a, b) - v_min(a, b);
+}
+
```

Comment:
templates ?

Reply:
I can't apply templates because there are v_absdiffs() function with different implementation. For example:

inline v_int8x16 v_absdiffs(const v_int8x16& a, const v_int8x16& b)
{
    v_int8x16 d = a - b;
    v_int8x16 m = a < b;
    return (d ^ m) - m;
}